src/dev-mock-service/http-mock-service/index.ts

```diff
@@ -0,0 +1,72 @@
+import type { HttpOverride } from "./types.js";
+import type { HandlerOverride } from "../types/index.js";
+import type { JSONValue } from "@devvit/public-api/public-api.js";
+
+export const createDevvFetch = (
+  realFetch: typeof fetch,
+  overrides: HttpOverride[],
+): typeof fetch => {
+  if (!overrides.length) {
+    return realFetch;
+  }
+  return (url, request) => {
+    const matchingOverride = overrides.find((override) => {
+      return (
+        override.url === url &&
+        override.method === (request?.method ?? "").toUpperCase()
+      );
+    });
+
+    if (!matchingOverride) {
+      return realFetch(url, request);
+    }
+    const mockResponse = matchingOverride.handler(request);
+    return Promise.resolve(mockResponse);
+  };
+};
+
+type HttpHandlerFactory = (
```

Comment:
I don't really know what a handler is. is it the meat of the mock? should we call it HttpMockFactory? it would be great also if there wasn't overlap with "override".

Reply:
Yeah, I think I need to revisit the naming, about half of the code here was written 6 months ago 😅